src/request.m

```diff
@@ -295,8 +295,11 @@ - (void) _updateWindow: (NSInteger) delta withBlock: (void (^)()) block {
       window_out_queue_ = [NSMutableArray arrayWithCapacity: 16];
 
     // Retry on positive window_out
+    __weak __typeof(self) weakSelf = self;
```

Comment:
The above assumptions are based only by looking at the code from `_updateWindow`, as I'm not familiar with the ISpdy design pardon any naive assumptions :)